Task/update deps documentation (#3)
open-obfuscator/doc: omvll/introduction/compilation/index.md

##### On Linux

```bash {hl_lines=2}
export NDK_STAGE1=$(LLVM_ROOT)/out/stage1-install

> export NDK_STAGE2=$(LLVM_ROOT)/out/stage2

Reviewer: ```suggestion
export NDK_STAGE1=${LLVM_ROOT}/out/stage1-install
export NDK_STAGE2=${LLVM_ROOT}/out/stage2
```

cp ${NDK_STAGE2}/bin/clang ./tmp/bin

> cp ${NDK_STAGE2}/bin/clang++ ./tmp/bin

Reviewer: Meant `/tmp/bin`? Where was `bin` created? Is this necessary?

Author: No need of this two variables, I will remove this

$ cmake -GNinja ..                                              \
        -DCMAKE_BUILD_TYPE=Release                              \
        -DCMAKE_CXX_COMPILER=${NDK_STAGE1}/bin/clang++          \
        -DCMAKE_C_COMPILER=${NDK_STAGE1}/bin/clang              \
        -DCMAKE_CXX_FLAGS="-stdlib=libc++"                      \
        -DPython3_ROOT_DIR=${PYTHON_ROOT}                       \
        -DPython3_LIBRARY=${PYTHON_ROOT}/lib/libpython3.10.a    \
        -DPython3_INCLUDE_DIR=${PYTHON_ROOT}/include/python3.10 \
        -Dpybind11_DIR=${PYBIND11_ROOT}/share/cmake/pybind11    \
        -Dspdlog_DIR=${SPDLOG_ROOT}/lib/cmake/spdlog            \
        -DLLVM_DIR=${NDK_STAGE2}/lib64/cmake/llvm
$ ninja
```

##### On OSX

```bash {hl_lines=3}
$ CC=/Applications/Xcode15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang \
  CXX=/Applications/Xcode15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang \
  ++cmake -GNinja                                                 \
          -S src                                                  \
          -B build                                                \

>           -DCMAKE_OSX_DEPLOYMENT_TARGET="13.0"                    \

Reviewer: ```suggestion
$ cmake   -GNinja                                                 \
          -S src                                                  \
          -B build                                                \
          -DCMAKE_C_COMPILER=$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang              \
          -DCMAKE_CXX_COMPILER=$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++            \
          -DCMAKE_OSX_DEPLOYMENT_TARGET="13.0"                    \
```

the pre-compiled dependencies).
{{</ admonition >}}

> {{< hicon lvl=2 icon="fa-brands fa-python" >}}Generate NDK deps{{< /hicon >}}

Reviewer: Nice we script all of this now :)

> O-MVLL dependencies can be generated easily using same O-MVLL respository:

Reviewer: ```suggestion
O-MVLL dependencies can be generated easily using the same repository:
```

```bash
# Make sure you have expected ANDROID_HOME variable is already satisfied
ANDROID_HOME="sdk folder installation"
# Install ndk version used by the project
$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager --install 'ndk;25.2.9519653'

> git clone https://github.com/open-obfuscator/o-mvll.git

Reviewer: ```suggestion
git clone https://github.com/open-obfuscator/o-mvll.git o-mvll && cd o-mvll
```

./scripts/deps/generate_deps.sh -p ndk -o output.tar.gz

> # output.tar.gz will generated with all required dependencies

Reviewer: ```suggestion
# output.tar.gz will be generated with all the required dependencies.
```

```

{{< hicon lvl=2 icon="fa-light fa-input-text" >}}Generate XCode deps{{< /hicon >}}

> O-MVLL dependencies can be generated easily using same O-MVLL respository:

Reviewer: ```suggestion
O-MVLL dependencies can be generated easily using the same repository:
```

```bash

> git clone https://github.com/open-obfuscator/o-mvll.git

Reviewer: ```suggestion
git clone https://github.com/open-obfuscator/o-mvll.git o-mvll && cd o-mvll
```

./scripts/deps/generate_deps.sh -p xcode -o output.tar.gz

> # output.tar.gz will generated with all required dependencies

Reviewer: ```suggestion
# output.tar.gz will be generated with all the required dependencies.
```
